lib/resque_bus/tasks.rb

```diff
@@ -1,106 +1,128 @@
 # require 'resque_bus/tasks'
 # will give you these tasks
 
-require "queue_bus/tasks"
-require "resque/tasks"
+require 'queue_bus/tasks'
+require 'resque/tasks'
 
 namespace :resquebus do
-  # deprecated
-  task :setup => ["queuebus:setup"] do
-    ResqueBus.note_deprecation "[DEPRECATION] rake resquebus:setup is deprecated. Use rake queuebus:setup instead."
+
+  task setup: ['queuebus:setup'] do
+    ResqueBus.note_deprecation '[DEPRECATION] rake resquebus:setup is deprecated. Use rake queuebus:setup instead.'
   end
 
-  task :driver => ["queuebus:driver"] do
-    ResqueBus.note_deprecation "[DEPRECATION] rake resquebus:driver is deprecated. Use rake queuebus:driver instead."
+  task driver: ['queuebus:driver'] do
+    ResqueBus.note_deprecation '[DEPRECATION] rake resquebus:driver is deprecated. Use rake queuebus:driver instead.'
   end
 
-  task :subscribe => ["queuebus:subscribe"] do
-    ResqueBus.note_deprecation "[DEPRECATION] rake resquebus:subscribe is deprecated. Use rake queuebus:subscribe instead."
+  task subscribe: ['queuebus:subscribe'] do
+    ResqueBus.note_deprecation '[DEPRECATION] rake resquebus:subscribe is deprecated. Use rake queuebus:subscribe instead.'
   end
 
-  task :unsubsribe => ["queuebus:unsubsribe"] do
-    ResqueBus.note_deprecation "[DEPRECATION] rake resquebus:driver is deprecated. Use rake queuebus:unsubsribe instead."
+  task unsubsribe: ['queuebus:unsubsribe'] do
+    ResqueBus.note_deprecation '[DEPRECATION] rake resquebus:driver is deprecated. Use rake queuebus:unsubsribe instead.'
   end
 end
 
+namespace :resque do
+  Rake::Task[:work].enhance ['queuebus:set_redis_server']
+end
+
 namespace :queuebus do
 
-  desc "Setup will configure a resque task to run before resque:work"
-  task :setup => [ :preload ] do
+  desc 'Sets the resque worker to use the QueueBus Redis server'
+  task :set_redis_server do
+    unless ENV['USE_QUEUEBUS_REDIS'].nil?
```

Comment:
I think it will leave room for "Ooops! I forgot to set / unset the other variable" error. We can just unset QUEUEBUS_REDIS_URL if we want queuebus to use the default redis instance.